File: app/src/main/java/br/com/brqtest/deserializers/CEPDeserializer.java
package br.com.brqtest.deserializers;

import br.com.brqtest.model.Endereco;
import com.google.gson.*;

import java.lang.reflect.Type;

public class CEPDeserializer implements JsonDeserializer {

    @Override
    public Object deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context) throws JsonParseException {

        JsonElement element = json.getAsJsonObject();

        if(json.getAsJsonObject() != null){
            element = json.getAsJsonObject();
        }

        return (new Gson().fromJson(element, Endereco.class));
    }
}

File: app/src/main/java/br/com/brqtest/deserializerskotlin/CEPDeserializer.kt
package br.com.brqtest.deserializerskotlin

import br.com.brqtest.modelkotlin.Endereco
import com.google.gson.*
import java.lang.reflect.Type

class CEPDeserializer : JsonDeserializer<Any> {

    @Throws(JsonParseException::class)
    override fun deserialize(json: JsonElement, typeOfT: Type, context: JsonDeserializationContext): Any {

        var element: JsonElement = json.asJsonObject

        if (json.asJsonObject != null) {
            element = json.asJsonObject
        }

        return Gson().fromJson(element, Endereco::class.java)
    }
}